### ISSUE TITLE
Updates方法如果更新字段不在model属性列表中,则会跳过该字段的更新并不会报错

### ISSUE THREAD

**Issue by lukexwang** [go-gorm/gorm#2297]
Your issue may already be reported! Please search on the [issue track](https://github.com/jinzhu/gorm/issues) before creating one.

### What version of Go are you using (`go version`)?
go version go1.10.1 linux/amd64

### Which database and its version are you using?
MySQL 7.2.25

### Please provide a complete runnable program to reproduce your issue. **IMPORTANT**

Need to runnable with [GORM's docker compose config](https://github.com/jinzhu/gorm/blob/master/docker-compose.yml) or please provides your config.

```go
type BillCrontabSync struct {
	Uid int `json:"uid" gorm:"column:uid;AUTO_INCREMENT;"`
	BillUid int `json:"billuid" gorm:"column:billuid;type:int"`
	StartTime string `json:"start_time" gorm:"column:start_time;type:datetime;not null" binding:"required"`
	User string `json:"user" gorm:"column:user;type:varchar(128)" binding:"required"`
	APP string `json:"app" gorm:"column:app;type:varchar(128);not null" binding:"required"`
	DBIpList string `json:"dbiplist" gorm:"column:dbiplist;type:mediumtext"`
	DRIpList string `json:"driplist" gorm:"column:driplist;type:mediumtext"`
	DBs string `json:"dbs" gorm:"column:dbs;type:mediumtext"`
	IgnoreDBs string `json:"ignore_dbs" gorm:"column:ignore_dbs;type:mediumtext"`
	Tables string `json:"tables" gorm:"column:tables;type:mediumtext"`
	IgnoreTables string `json:"ignore_tables" gorm:"column:ignore_tables;type:mediumtext"`
	SpecTables string `json:"spec_tables" gorm:"column:spec_tables;type:mediumtext"`
	SpecIgnoreTables string `json:"spec_ignore_tables" gorm:"column:spec_ignore_tables;type:mediumtext"`
	SyncNonInnodbTbls int `json:"sync_non_innodb_tbls" gorm:"column:sync_non_innodb_tbls;type:tinyint"`
	SyncResult string `json:"sync_result" gorm:"column:sync_result;type:mediumtext"`
}

func UpdateSyncResult(job *BillCrontabSync,undoSubNum int,doingSubNum int,doneSubNum int,syncResult string) error{
	log.Logger.Info("UpdateSyncResult",zap.Int("undoSubNum",undoSubNum),zap.Int("doingSubNum",doingSubNum),zap.Int("doneSubNum",doneSubNum))
	if err:=DB.TestDB.Model(BillCrontabSync{}).Where("uid=?",job.Uid).
		UpdateColumns(map[string]interface{}{
			"sync_result":gorm.Expr("concat(sync_result,?)",syncResult),
			"undo_num":gorm.Expr("undo_num - ?",undoSubNum),
			"doing_num":gorm.Expr("doing_num - ?",doingSubNum),
			"done_num":gorm.Expr("done_num - ?",doneSubNum),
			}).Error;err!=nil{
		log.Logger.Error("update sync_result fail",zap.Int("uid",job.Uid),zap.String("syncResult",syncResult),zap.Error(err))
		return err
	}
	return nil
}
```
运行结果只更新了 sync_result 一个字段。 日志打印的SQL:
```SQL
UPDATE `bill_crontab_sync` SET `sync_result` = concat(sync_result,'DR:127.0.0.1#20001 some unknown errors occured
')  WHERE (uid='1') 
```
附上参考文章:[记一次 Gorm.Updates 引发的惨案](https://tyrodw.github.io/post/golang/gorm-updates/)

**Comment by ruolinn** [comment]
我也遇见了同样的问题 调试了一下午

**Comment by magicdvd** [comment]
+1

**Comment by Xuhao** [comment]
@jinzhu   同样遇到这个bug，算是挺验证的bug了，不知道现在有什么进展吗？

**Comment by jinzhu** [comment]
Fixed, please upgrade gorm to v1.9.13.

BTW, GORM v2 should have this fixed by default. Checkout https://github.com/go-gorm/gorm/wiki/GORM-V2-Release-Note-Draft-CN for how to upgrade.

**Comment by Xuhao** [comment]
谢谢，很期待 v2 😍
